data-prepper-plugins/s3-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceTest.java

```diff
@@ -0,0 +1,198 @@
+        assertNotNull(s3Client);
+        assertThat(s3Client, instanceOf(S3Client.class));
+    }
+
+    @Test
+    void test_generateKey_with_general_prefix() {
+        String pathPrefix = "events/";
+        when(s3SinkConfig.getBucketOptions().getObjectKeyOptions().getPathPrefix()).thenReturn(pathPrefix);
+        S3SinkService s3SinkService = new S3SinkService(s3SinkConfig, bufferFactory, codec);
+        String key = s3SinkService.generateKey();
+        assertNotNull(key);
+        assertThat(key, true);
+        assertThat(key, key.contains(pathPrefix));
+    }
+
+    @Test
+    void test_generateKey_with_date_prefix() {
+        String pathPrefix = "logdata/";
+        String datePattern = "%{yyyy}/%{MM}/%{dd}/";
+
+        DateTimeFormatter fomatter = DateTimeFormatter.ofPattern("yyyy/MM/dd");
+        ZonedDateTime zdt = LocalDateTime.now().atZone(ZoneId.systemDefault())
+                .withZoneSameInstant(ZoneId.of(TimeZone.getTimeZone("UTC").getID()));
+        String dateString = fomatter.format(zdt);
+
+        when(s3SinkConfig.getBucketOptions().getObjectKeyOptions().getPathPrefix()).thenReturn(pathPrefix + datePattern);
+        S3SinkService s3SinkService = new S3SinkService(s3SinkConfig, bufferFactory, codec);
+        String key = s3SinkService.generateKey();
+        assertNotNull(key);
+        assertThat(key, true);
+        assertThat(key, key.contains(pathPrefix + dateString));
+    }
+
+    @Test
+    void test_output_with_threshold_set_as_more_then_zero_event_count() throws IOException {
+        when(s3SinkConfig.getThresholdOptions().getEventCount()).thenReturn(10);
+        when(codec.parse(any())).thenReturn("{\"message\":\"31824252-adba-4c47-a2ac-05d16c5b8140\"}");
+        S3SinkService s3SinkService = new S3SinkService(s3SinkConfig, bufferFactory, codec);
+        assertNotNull(s3SinkService);
+        s3SinkService.output(generateRandomStringEventRecord());
+    }
+
+
+    // If event_count threshold set as zero, Hence event_count will be
+    // ignored as part of threshold check.
+    @Test
+    void test_output_with_threshold_set_as_zero_event_count() throws IOException {
+        when(s3SinkConfig.getThresholdOptions().getEventCount()).thenReturn(0);
+        when(s3SinkConfig.getThresholdOptions().getMaximumSize()).thenReturn(ByteCount.parse("2kb"));
+        when(codec.parse(any())).thenReturn("{\"message\":\"31824252-adba-4c47-a2ac-05d16c5b8140\"}");
+        S3SinkService s3SinkService = new S3SinkService(s3SinkConfig, bufferFactory, codec);
+        assertNotNull(s3SinkService);
+        s3SinkService.output(generateRandomStringEventRecord());
+    }
+
+    @Test
+    void test_catch_output_exception() {
+        S3SinkService s3SinkService = new S3SinkService(s3SinkConfig, bufferFactory, codec);
+        assertNotNull(s3SinkService);
+        s3SinkService.output(generateRandomStringEventRecord());
+        assertThat(s3SinkService, instanceOf(S3SinkService.class));
```

Comment:
Please move this one line above. No point to asserting this after doing output().